github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/GitHubPRTrigger.java

```diff
@@ -336,6 +343,31 @@ private static boolean isSupportedTriggerMode(GitHubPRTriggerMode mode) {
         }
     }
 
+    private static interface PullRequestSupplier<T> {
+        T getPullRequests() throws IOException;
+    }
+
+    private static <T> T getPullRequests(PullRequestSupplier<T> supplier) throws IOException {
+        T pullRequests = null;
+
+        int count = 0;
+        int retries = 3;
+        while (count++ < retries) {
+            try {
+                pullRequests = supplier.getPullRequests();
+                break;
+            } catch (IOException e) {
+                if (count == retries) {
+                    throw e;
+                }
+
+                LOGGER.debug("Failed retrieving pull request(s), retrying...", e);
```

Comment:
no, sleep is fine, Trigger is executed under Timer threads, if trigger needs time, then it needs it.